test/Microsoft.Windows.CsWin32.Tests/GeneratorTests.cs

```diff
@@ -2170,6 +2171,38 @@ static string JoinAssemblyMetadata(string name)
 
     private static IEnumerable<AttributeSyntax> FindAttribute(SyntaxList<AttributeListSyntax> attributeLists, string name) => attributeLists.SelectMany(al => al.Attributes).Where(a => a.Name.ToString() == name);
 
+    private static void AssertConsistentLineEndings(Compilation compilation)
+    {
+        foreach (SyntaxTree doc in compilation.SyntaxTrees)
+        {
+            AssertConsistentLineEndings(doc);
+        }
+    }
+
+    private static void AssertConsistentLineEndings(SyntaxTree syntaxTree)
```

Comment:
💡 You could use the new `EqualOrDiff` functionality from dotnet/roslyn-sdk for this assertion by asserting the `syntaxTree.GetText()` is the same as `syntaxTree.GetText().Replace("\r\n", "\n")`.

https://github.com/dotnet/roslyn-sdk/pull/886

Reply:
Ooh, that's nice. Thanks for the tip. Where is this `EqualOrDiff` method exposed, or would I need to copy its source?

Reply:
If you already reference MS.CA.Testing, you can use `DefaultVerifier` with [this extension method](https://github.com/dotnet/roslyn-sdk/blob/1c5cbe8d9add7cce50443620730124d60e714f2f/src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/Extensions/IVerifierExtensions.cs#L30). Otherwise the source is reasonably easy to copy.

Reply:
Note that `EqualOrDiff` was originally implemented by you, but we modified it in dotnet/roslyn-sdk#878 and dotnet/roslyn-sdk#886 to specifically support cases where only the line endings changed.

Reply:
Ya, I thought I remembered that I had shared it. We developed it for vs-mef.